mycore-base/src/main/java/org/mycore/frontend/cli/MCRCommandUtils.java

```diff
@@ -0,0 +1,169 @@
+package org.mycore.frontend.cli;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.mycore.common.MCRException;
+import org.mycore.common.MCRPersistenceException;
+import org.mycore.common.MCRUsageException;
+import org.mycore.common.config.MCRConfiguration2;
+import org.mycore.datamodel.common.MCRXMLMetadataManager;
+import org.mycore.datamodel.metadata.MCRMetadataManager;
+import org.mycore.datamodel.metadata.MCRObjectID;
+
+/**
+ * Utilities intended to reduce redundant code when writing variants of CLI commands.
+ * 
+ * @author Christoph Neidahl (OPNA2608)
+ *
+ */
+public class MCRCommandUtils {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    /**
+     * Get a stream of MCRObjectIDs found for a type.
+     * 
+     * @param type
+     *     The type to look in the store for.
+     * @return A (parallel) stream with found IDs. (may be empty if none were found)
+     * @throws MCRUsageException
+     *     If no type was given or the type could not be found.
+     *     Not thrown if type exists but has no values.
+     */
+    public static final Stream<String> getIdsForType(final String type) throws MCRUsageException {
+        if (type == null || type.length() == 0) {
+            throw new MCRUsageException("Type required to enumerate IDs!");
+        }
+        if (MCRConfiguration2.getString("MCR.Metadata.Type." + type).isEmpty()) {
+            throw new MCRUsageException(
+                "MCR.Metadata.Type." + type + " not set!");
+        }
+        List<String> idList;
+        try {
+            idList = MCRXMLMetadataManager.instance().listIDsOfType(type);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new MCRUsageException("Requested type " + type + " not found!", e);
```

Comment:
Do not throw an exception here! Never ever catch RuntimeException if you cannot handle them.